src/math_functions.cpp

```diff
@@ -0,0 +1,818 @@
+#include "math_functions.h"
+
+namespace openmc {
+
+//==============================================================================
+// Module constants.
+//==============================================================================
+
+// TODO: cmath::M_PI has 3 more digits precision than the Fortran constant we
+// use so for now we will reuse the Fortran constant until we are OK with
+// modifying test results
+extern "C" const double PI {3.1415926535898};
+
+extern "C" const double SQRT_PI {std::sqrt(PI)};
+
+//==============================================================================
+// NORMAL_PERCENTILE calculates the percentile of the standard normal
+// distribution with a specified probability level
+//==============================================================================
+
+double __attribute__ ((const)) normal_percentile_c(const double p) {
```

Comment:
`__attribute__ ((const))` is just for GCC right?  I'd rather we don't get married to just one compiler.  If you really think we need GCC-specific stuff then we should probably use it in a macro with `#ifdef`s so we can use other compilers (but I suspect optimizing these functions will make little difference to the overall runtime).

Reply:
Oh no you're right.  I do not intend to make GCC only optimizations; I was really looking for an equivalent to Fortran's `pure` identifier.  I've removed these